### PR TITLE
Show contribution details on member profile page

### PR DESCRIPTION
**Pull request by sumitd94**
Ref to issue #90 

This is how it looks

image

### CODE REVIEW

File: components/member-profile/contribution/index.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  } else {
+    const createdAt = +new Date(prList[0]['createdAt']);
+    const updatedAt = +new Date(prList[0]['updatedAt']);
+    if (prList[0]['state'] === 'closed') {
+      completedDate = timeWas(createdAt, false, updatedAt);
+      completedText = <span>Completed in: </span>;
+      featurLiveDate = timeWas(updatedAt, true);
+      featureLiveOnText = `Feature live on ${featurLiveDate}`;
+    }
+  }
```
Comment (sumitd94) [inline components/member-profile/contribution/index.js:52]: Is there a way to refactor this? I am not able to figure it out. Open to any sort of suggestion.

File: helper-functions/urls.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
 const baseURL = 'https://raw.githubusercontent.com/Real-Dev-Squad/website-static/main';
-const PRbaseURL = 'https://staging-api.realdevsquad.com/pullrequests/user';
+const PRbaseURL = 'https://staging-api.realdevsquad.com/contributions';
```
Comment (ankushdharkar) [inline helper-functions/urls.js:2]: NIT: variable name can be changed

File: components/member-profile/contribution/contribution.module.scss [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+}
+
+.heading {
+  display: flex;
+  justify-content: space-between;
+}
+
+.featureTitle,
+.featureLink a {
+  color: #041484 !important;
```
Comment (ankushdharkar) [inline components/member-profile/contribution/contribution.module.scss:21]: @sumitd94 Can we do away with the `!important` here?

File: components/member-profile/contribution/index.js [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+  const [showMoreContent, setShowMoreContent] = useState(false);
+
+  const showMoreContentHandler = () => {
+    setShowMoreContent((prevstate) => !prevstate);
+  };
+
+  const showText = showMoreContent ? 'Show less' : 'Show more';
+  const showMoreContentClass = showMoreContent ? classNames.showContent : classNames.hideContent;
+
+  const isTaskAvailable = title ? true : false;
```
Comment (ankushdharkar) [inline components/member-profile/contribution/index.js:26]: NIT: `const isTaskAvailable = !!title;`

File: components/member-profile/contribution/index.js (hunk 1 above)
Comment (ankushdharkar) [inline components/member-profile/contribution/index.js:52]: @sumitd94: Let's refactor this in the next PR? :)

File: components/member-profile/contribution/index.js [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+      completedText = <span>Completed in: </span>;
+      featurLiveDate = timeWas(updatedAt, true);
+      featureLiveOnText = `Feature live on ${featurLiveDate}`;
+    }
+  }
+
+  return (
+    <div className={classNames.contributionContainer}>
+      <section className={classNames.heading}>
+        <span className={classNames.featureTitle}>{isTaskAvailable ? title : prList[0].title}</span>
```
Comment (ankushdharkar) [inline components/member-profile/contribution/index.js:57]: @sumitd94 Maybe this needs a default name?

File: components/member-profile/contribution/index.js [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+        <div className={classNames.prLink}>{renderPRLinks(prList)}</div>
+      </section>
+      {isTaskAvailable && (
+        <div className={showMoreContentClass}>
+          <p className={classNames.featureDescription}>{purpose}</p>
+          <p className={classNames.userIcon}>
+            <img src={imageLink} className={classNames.participantIcon} alt="participantsIcon" />
+            <span>{fullName}</span>
+          </p>
+          <p className={classNames.featureLink}>
```
Comment (ankushdharkar) [inline components/member-profile/contribution/index.js:67]: Guard statement missing

File: components/member-profile/contribution/index.js [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+            <a href={featureUrl} target="_blank" rel="noreferrer">
+              Click here to checkout this feature
+            </a>
+          </p>
+        </div>
+      )}
+      <p>
```
Comment (ankushdharkar) [inline components/member-profile/contribution/index.js:74]: @sumitd94 Refactor

File: components/member-profile/contribution/index.js [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+      featurLiveDate = timeWas(endsOn * 1000, true);
+      featureLiveOnText = `Feature live on ${featurLiveDate}`;
+    }
+  } else {
+    const createdAt = +new Date(prList[0]['createdAt']);
+    const updatedAt = +new Date(prList[0]['updatedAt']);
+    if (prList[0]['state'] === 'closed') {
+      completedDate = timeWas(createdAt, false, updatedAt);
+      completedText = <span>Completed in: </span>;
+      featurLiveDate = timeWas(updatedAt, true);
```
Comment (ankushdharkar) [inline components/member-profile/contribution/index.js:49]: Typo

**Review by ankushdharkar — APPROVED** [review]
@sumitd94 Please create and work on the refactor PR

File: components/member-profile/contribution/index.js [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+
+const Contribution = ({ contribution, fullName, imageLink }) => {
+  const {
+    task: { title, startedOn, endsOn, purpose, featureUrl, status },
+    prList
+  } = contribution;
+  const [showMoreContent, setShowMoreContent] = useState(false);
+
+  const showMoreContentHandler = () => {
+    setShowMoreContent((prevstate) => !prevstate);
```
Comment (sumitd94) [inline components/member-profile/contribution/index.js:20]: @ankushdharkar  ref - https://reactjs.org/docs/hooks-reference.html#functional-updates

File: helper-functions/urls.js (hunk 2 above)
Comment (sumitd94) [reply helper-functions/urls.js:2]: #121

File: components/member-profile/contribution/index.js (hunk 4 above)
Comment (sumitd94) [reply components/member-profile/contribution/index.js:26]: Done, Thanks.
#121

File: components/member-profile/contribution/index.js (hunk 1 above)
Comment (sumitd94) [reply components/member-profile/contribution/index.js:52]: Done.
#121

File: components/member-profile/contribution/index.js (hunk 6 above)
Comment (sumitd94) [reply components/member-profile/contribution/index.js:67]: #121

File: components/member-profile/contribution/index.js (hunk 7 above)
Comment (sumitd94) [reply components/member-profile/contribution/index.js:74]: #121

File: components/member-profile/contribution/index.js (hunk 8 above)
Comment (sumitd94) [reply components/member-profile/contribution/index.js:49]: #121

**Pull request merged**